src/tournament_commands/player_commands/match_result.rs

```diff
@@ -26,7 +27,13 @@ async fn match_result(ctx: &Context, msg: &Message, mut args: Args) -> CommandRe
 
        let tourns = data.get::<TournamentMapContainer>().unwrap();
        let user_name = msg.author.id;
-       let match_result = args.single_quoted::<String>().unwrap();
+       let player_id = tourn.players.get_right(&user_name).unwrap().clone();
+       //i have no idea if the RoundResult implementation is actually correct, the compiler doesn't seem to complain so this is what im rolling with
+       let result_of_match = match args.single_quoted::<String>().unwrap() {
```

Comment:
This is not how we are recording match results. Instead, in every round, players input how many games they have won and if there were any draws (wins and draws are separate evocations). So, a command like `!tournament match-result 2` would mean that I won two games. Or, `!tournament match-result draw` would mean that were was a draw in my pairing.

There is an admin equivalent to this command `admin_commands/match_result.rs` that has the correct parsing, but the key piece is:
```suggestion
let raw_result = args.single::<String>().unwrap();
// Some other arg parsing
let round_result = match raw_result.parse::<u8>() {
    Ok(n) => RoundResult::Wins(plyr_id, n),
    Err(_) => {
        if raw_result == "draw" {
            RoundResult::Draw()
        } else {
            msg.reply(
                &ctx.http,
                "The third argument must be the number of wins for the player.",
            )
            .await?;
            return Ok(());
        }
    }
};
```